test/tic_tac_toe_test/board_test.exs

```diff
@@ -8,24 +8,43 @@ defmodule TicTacToe.BoardTest do
     assert BoardInspect.get(%Board{board_state: [1, 2, 3]}, 1) == 1
   end
 
+  test "gets the board's position count and dimension" do
+    assert BoardInspect.size(%Board{board_state: List.duplicate(nil, 9)}) ==
+             {9, 3}
+  end
+
+  test "when there is a winner, status is win and winner is token" do
+    outcome =
+      %Board{board_state: ["X", "X", "X", nil, nil, nil, nil, nil, nil]}
+      |> BoardInspect.outcome()
+
+    assert outcome == {:win, "X"}
+  end
+
+  test "when there is a draw, status is draw amd winner is nil" do
+    outcome =
+      %Board{board_state: ["X", "X", "O", "O", "X", "X", "X", "O", "O"]}
+      |> BoardInspect.outcome()
+
+    assert outcome == {:draw, nil}
+  end
+
+  test "when there is no winner or draw, status is :in_progress and winner is nil" do
+    outcome =
+      %Board{board_state: ["O", "X", "X", nil, nil, nil, nil, nil, nil]}
+      |> BoardInspect.outcome()
```

Comment:
Nice!

Reply:
One thing you could do to further insulate your tests from the details of the struct state would be to use the API to put thee board in the right state.

```suggestion
    outcome =
      Board.empty()
      |> BoardUpdate.place_token(2, "X")
      |> BoardUpdate.place_token(1, "O")
      |> BoardUpdate.place_tokeen(3, "X")
      |> BoardInspect.outcome()
```

Maybe something to consider as you refactor. YMMV.